adr/content-management/2023-01-06-upload-thumbnail-api-concept.md

```diff
@@ -0,0 +1,72 @@
+# 2023-01-06 - Upload Thumbnails API
+
+## Context
+We’re building an App that should optimize Thumbnails via the App. Currently, it’s not possible to upload Thumbnails via the API. All Thumbnails are processed by Shopware itself. There’s no way to upload your optimized Thumbnails via the API.
+
+However: Since it’s an App, we need to be able to do so via the API.
+
+## Decision
+### Migration
+Add a new column `isUploaded` into the media_thumbnail table. This flag lets us know if the thumbnail is uploaded or generated.
+
+### Core
+
+Add a new `updloadThumbnail` API in Shopware\Core\Content\Media\Api\MediaUploadController
+```php
+/**
+ * @Route("/api/_action/media/{mediaId}/upload-thumbnail", name="api.action.media.upload-thumbnail", methods={"POST"})
+ */
+public function uploadThumbnail(string $mediaId, Request $request, Context $context, ResponseFactoryInterface $responseFactory): Response
+{
+    // Based on mediaId, validate uploaded thumbnails has to map with the media thumbnail sizes
```

Comment:
As you uploader provides the images, we should just trust him with the image size and don't do other checks here anymore

Reply:
If we don't check the thumbnail size, how do they know which sizes are configured in the media folder of that media? or we just don't care about it, just allow them to upload whatever? I'm not sure this point 🤔

Reply:
We should not open the uploaded file to determine the image sizes. The client has to provide this and we just use it. That was the intention :)

Reply:
Then, the question is, how do we know which file is for which thumbnail size?
E.g: The client uploads a file, and I will not know it is for 400x400 or 800x800, .etc...
Do you have any idea for this case?